File: csharp/acronym/Acronym.cs
using System;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        string acronym = "";
        var listOfWord = phrase.Replace("-", " ").Replace("_", " ").Split(" ");

        foreach (var word in listOfWord)
        {
            if (word.Length > 0 && char.IsLetter(word[0]))
            {
                acronym += word[0].ToString().ToUpper();
            }
        }

        return acronym;
    }
}

File: java/acronym/src/main/java/Acronym.java
import java.util.Arrays;
import java.util.stream.Collectors;

class Acronym {
  private String phrase;

  Acronym(String phrase) {
    this.phrase = phrase;
  }

  String get() {
    return Arrays.asList(this.phrase.replaceAll("_", "").replaceAll("-", " ").split(" ")).stream()
        .filter(word -> word != " " && word != "")
        .map(word -> Character.toString(word.charAt(0)).toUpperCase())
        .collect(Collectors.joining(""));
  }
}
